### PR TITLE
remove stackdriver reference

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/istio/istio.io/pull/15123

cc @kyessenov @lei-tang

### CODE REVIEW

**Comment by kfaseela** [comment]
why is lint still failing? is there any more reference which is to be removed?

**Comment by zirain** [comment]
wait https://github.com/istio/api/pull/3204

**Review by lei-tang — APPROVED** [review]

File: data/features.yaml [hunk 1]
```diff
@@ -142,14 +142,6 @@ features:
       maturity: Stable
       nextExpectedPromotion: ""
     area: Observability
-  - name: "Stackdriver Integration"
```
Comment (ericvn) [inline data/features.yaml:145]: This change needs to be made in the istio/enhancements repo as it's pulled from there: 

https://github.com/istio/istio.io/blob/master/scripts/grab_reference_docs.sh#L114

**Review by ericvn — CHANGES_REQUESTED** [review]

File: data/features.yaml (hunk 1 above)
Comment (zirain) [reply data/features.yaml:145]: https://github.com/istio/enhancements/pull/179

**Pull request closed without merging**